frontend/src/components/widgets/DataGrid/DataGrid.tsx

```diff
@@ -42,73 +145,47 @@ function DataGrid({
-    fontFamily: theme.fonts.sansSerif,
-    bgSearchResult: transparentize(theme.colors.primary, 0.9),
-    // Header styling:
-    bgIconHeader: theme.colors.fadedText60,
-    fgIconHeader: theme.colors.white,
-    bgHeader: theme.colors.bgMix,
-    bgHeaderHasFocus: theme.colors.secondaryBg,
-    bgHeaderHovered: theme.colors.bgMix, // no hovering color change
-    textHeader: theme.colors.fadedText60,
-    textHeaderSelected: theme.colors.white,
-    headerFontStyle: `bold ${theme.fontSizes.sm}`,
-    // Cell styling
-    baseFontStyle: theme.fontSizes.sm,
-    editorFontSize: theme.fontSizes.sm,
-    textDark: theme.colors.bodyText,
-    textMedium: theme.colors.fadedText60,
-    textLight: theme.colors.fadedText40,
-    textBubble: theme.colors.fadedText60,
-    bgCell: theme.colors.bgColor,
-    bgCellMedium: theme.colors.bgColor, // TODO(lukasmasuch): Do we need a different color here?
-    cellHorizontalPadding: 8,
-    cellVerticalPadding: 3,
-    // Special cells:
-    bgBubble: theme.colors.secondaryBg,
-    bgBubbleSelected: theme.colors.secondaryBg,
-    linkColor: theme.colors.linkText,
-    drilldownBorder: "rgba(0, 0, 0, 0)", // TODO(lukasmasuch): Do we need a different color here?
-  }
+  const { numRows, columns, getCellContent } = useDataLoader(element)
+
+  // Automatic height calculation: numRows +1 because of header, and +3 pixels for borders
+  const height = propHeight || Math.min((numRows + 1) * ROW_HEIGHT + 3, 400)
 
-  // TODO(lukasmasuch): Apply automatic height calculation instead of the default value in later PRs
-  const height = propHeight || 300
+  // Calculate min height for the resizable container. header + one column, and +3 pixels for borders
+  const minHeight = 2 * ROW_HEIGHT + 3
 
   return (
-    // This is a styled-components theme provider (not emotion!).
-    // It is required by glide-data-grid to customize the theming.
-    <ThemeProvider theme={dataGridTheme}>
-      <DataGridContainer
-        className="stDataGrid"
-        width={width}
-        height={height}
-        theme={theme}
-      >
-        <GlideDataEditor
-          // TODO(lukasmasuch): This is currently only mock implementations.
-          // The actual implementations will be added in a later PR.
-          getCellContent={([col, row]: readonly [
-            number,
-            number
-          ]): GridCell => ({
-            kind: GridCellKind.Text,
-            data: "",
-            displayData: "",
-            allowOverlay: true,
-          })}
-          columns={[]}
-          rows={0}
-        />
-      </DataGridContainer>
-    </ThemeProvider>
+    <ThemedDataGridContainer
+      width={width}
+      height={height}
+      minHeight={minHeight}
+    >
+      <GlideDataEditor
+        // Callback to get the content of a given cell location:
+        getCellContent={getCellContent}
+        // List with column configurations:
+        columns={columns}
+        // Number of rows:
```

Comment:
I think a lot of these comments are not necessary. I think the variable names are probably explanatory imo.